### PR TITLE
fixed karb parser

### PR DESCRIPTION
**Pull request by librarywebchic**

### CODE REVIEW

File: karb/parser.js [hunk 1]
```diff
@@ -51,14 +57,19 @@ module.exports = new Parser(function analyseEC(parsedUrl, ec) {
...
-      result.rtype  = undefined;
+    if ((param.id && param.id.includes('Ch')) || (param.ids && param.ids.includes('Ch'))) {
+      if ((param.id && param.id.match(/Ch/g).length > 1) || (param.ids && param.ids.match(/Ch/g).length > 1)) {
+        result.rtype  = 'BOOK_CHAPTERS_BUNDLE';
+      } else {
+        result.rtype  = 'BOOK_SECTION';
+      }
+    } else if (param.id || param.ids) {
+      result.rtype  = 'ARTICLE';
       result.unitid = param.ids;
```
Comment (wilmouths) [inline karb/parser.js:70]: According to your condition, the unitid should be ``result.unitid = param.id || param.ids``

File: karb/parser.js [hunk 2]
```diff
@@ -74,7 +85,11 @@ module.exports = new Parser(function analyseEC(parsedUrl, ec) {
     // /document/kli-ka-ai-2020-01-002?title=Arbitration%20International
     // /document/KLI-KCC-1103103-n
 
-    result.rtype  = param.title ? 'ARTICLE' : 'BOOK_SECTION';
+    if (param.title && param.title != 'PDF') {
```
Comment (wilmouths) [inline karb/parser.js:88]: Prefer to use a strict inequality for string comparison. ``param.title !== 'PDF'``

File: karb/parser.js [hunk 3]
```diff
@@ -27,16 +27,22 @@ module.exports = new Parser(function analyseEC(parsedUrl, ec) {
...
       result.unitid = param.ids;
-      result.rtype  = 'ARTICLE';
       result.mime   = 'PDF';
     }
+
     if (param.journal) {
       result.unitid = param.journal;
     }
+
+    if ((param.id && param.id.includes('Ch')) || (param.ids && param.ids.includes('Ch'))) {
```
Comment (wilmouths) [inline karb/parser.js:40]: Do you have an example for this case, to rule out any concerns about side effects and find a solution perhaps more suitable.

File: karb/parser.js [hunk 4]
```diff
@@ -3,13 +3,33 @@
 'use strict';
 const Parser = require('../.lib/parser.js');
 
+/**
+ * Extract rtype
+ */
+function determineRtype (param, result) {
+  let idMatch;
+  let ids = param.id || param.ids;
```
Comment (wilmouths) [inline karb/parser.js:11]: Prefer use ``const`` in this case

File: karb/parser.js [hunk 5]
```diff
@@ -3,13 +3,33 @@
...
+  idMatch = /^(.+)Ch[0-9]+$/.exec(ids);
+  let idArray = ids.split(',');
```
Comment (wilmouths) [inline karb/parser.js:13]: Prefer use ``const`` in this case

File: karb/parser.js [hunk 6]
```diff
@@ -3,13 +3,33 @@
 'use strict';
 const Parser = require('../.lib/parser.js');
 
+/**
+ * Extract rtype
+ */
+function determineRtype (param, result) {
+  let idMatch;
```
Comment (wilmouths) [inline karb/parser.js:10]: Not necessary in this case, prefer write ``const idMatch = /^(.+)Ch[0-9]+$/.exec(ids);`` in line 12

File: karb/parser.js (hunk 4 above)
Comment (librarywebchic) [reply karb/parser.js]: updates made

**Pull request merged**